### DISCUSSION TITLE
Release Candidate 3.5.1 RC0

### DISCUSSION THREAD

**Discussion by ynaessens** [JeromeDevome/GRR#219] [General]
Cette branche apporte deux modifications majeures : 
- les champs login, etc. sont allongés pour compatibilité avec certains SSO,
- le cryptage des mots de passe est changé de md5 à password_hash().

Sauf retours négatifs, cette version sera proposée en stable d'ici à trois semaines.

<hr /><em>This discussion was created from the release <a href='https://github.com/JeromeDevome/GRR/releases/tag/v3.5.1RC0'>Release Candidate 3.5.1 RC0</a>.</em>

**Comment by zpaul91** [comment]
Bonjour
La version GRR 3.5.1 va t elle prochainement passer à stable ? 
Merci.

**Reply by ynaessens** [reply]
Bonjour,
apparemment il y a un souci avec l’authentification en liaison avec un annuaire AD, en cours d’investigation.
Dès que possible sortira une RC1 pour confirmation.
Cordialement,
YN

**Reply by zpaul91** [reply]
Bonjour, 
En effet, je viens de voir le problème, quand c'est la première connexion via LDAP, aucun pbl, mais quand on est déjà présent dans la base, ça ne fonctionne pas. 
La 3.5.0 est-elle plus stable ? 
La 4.0.0 est elle compatible PHP8.1 ? Ou pour une compatibilité PHP8.1 il faut rester sur une version de branche 3.5 ? 
Merci à vous,

**Reply by ynaessens** [reply]
Bonjour,
merci pour ces informations.
Pour ce qui est de la stabilité de la v3.5.0, je pourrais la confirmer si vous la testiez avec succès ;-)
En principe la v4.0.0 est également compatible avec php8.1, mais là aussi les retours d'expérience avec les connexions LDAP sont bienvenues !
Cordialement,
YN

**Reply by zpaul91** [reply]
Bonjour, 
Je viens de vous faire une pull request, pour corriger l'auth LDAP, SSO et IMAP, à partir de la 3.5.1
J'en ai profité pour désactiver SE3 quand le param n'est pas défini.
Je vous laisse regarder et valider la pull request si OK 
Bonne journée

**Reply by ynaessens** [reply]
Bonjour,
je viens d'étudier votre PR. Je ne comprends pas pourquoi vous ajoutez la condition source = 'local' dans la requête puisqu'on cherche à ce point un utilisateur à identifier en LDAP ou IMAP. D'accord pour la désactivation de SE3, dont je doute aujourd'hui que ce soit encore utile.
Merci pour votre contribution et vos explications à venir ;-)

**Reply by zpaul91** [reply]
Rebonjour, 
La requete cherche à vérifier si on a un utilisateur qui correspond à ce login dans la base des utilisateurs.
Si on regarde la suite, si pas de résultat, on tente l'authentification extérieure (LDAP, SSO, IMAP, ...) 
Sinon, on appelle le checkPassword, donc l'utilisateur est reconnu comme "local", avec un password. 

Cela semble venir de l'introduction du hash password, où on ne peut pas simplement faire la requete pour tenter de trouver un utilisateur qui répond au couple login/pass directement en base. Quand on avait une correspondance, on tentait l'auth LDAP/SSO/IMAP : c'est le comportement que j'ai voulu remettre en place. Il y a sans doute autre méthode, mais ça fonctionne sous LDAP (donc, ça devrait aussi fonctionner sous SSO etc. mais je n'ai pas le moyen de tester le SSO).

**Reply by ynaessens** [reply]
Je vois. 
L'erreur que j'avais faite était donc de rechercher l'utilisateur en base sans tester la source ?
Ensuite, la deuxième connexion se passe bien ?
Pas de souci de casse majuscule/minuscule sur les identifiants ? J'ai un doute sur l'introduction de strtoupper()...

**Reply by zpaul91** [reply]
Je pense que c'est l'erreur oui, car par défaut si on trouve l'utilisateur, alors on passe à la vérification des passwords et de l'inactivité du compte.
La deuxième connexion se passe bien, aussi la première connexion.
Pour l'introduction de strtoupper, pas de problème : UPPER(login) login met le champ login en majuscule, donc, la condition WHERE vérifie majuscule vs majuscule.

**Reply by ynaessens** [reply]
Un grand merci !
Je reprends votre proposition.
Bonne journée à vous aussi.
